packages/protocol/test-sol/governance/voting/ReleaseGold.t.sol

```diff
@@ -0,0 +1,2208 @@
+    vm.prank(releaseOwner);
+    releaseGold.revoke();
+  }
+}
+
+contract Expire is ReleaseGoldTest {
+  function setUp() public {
+    super.setUp();
+    newReleaseGold(true, false);
+  }
+
+  function test_ShouldRevert_WhenCalledBeforeExpirationTimeHasPassed() public {
+    vm.expectRevert("`EXPIRATION_TIME` must have passed after the end of releasing");
+    vm.prank(releaseOwner);
+    releaseGold.expire();
+  }
+
+  function test_ShouldRevertBeforeEXPIRATION_TIMEAfterReleaseScheduleEnd_WhenTheContractHasFinishedReleasing()
+    public
+  {
+    (, , uint256 numReleasePeriods, uint256 releasePeriod, ) = releaseGold.releaseSchedule();
+    vm.warp(block.timestamp + numReleasePeriods * releasePeriod + 5 * MINUTE);
+    vm.expectRevert("`EXPIRATION_TIME` must have passed after the end of releasing");
+    vm.prank(releaseOwner);
+    releaseGold.expire();
+  }
+
+  function test_ShouldRevert_WhenNotCalledByReleaseOwner_WhenEXPIRATION_TIMEHasPassedAfterReleaseScheduleCompletion()
+    public
+  {
+    (uint256 releaseStartTime, , uint256 numReleasePeriods, uint256 releasePeriod, ) = releaseGold
+      .releaseSchedule();
+    vm.warp(
+      releaseStartTime + numReleasePeriods * releasePeriod + releaseGold.EXPIRATION_TIME() + 1
+    );
+    vm.expectRevert("Sender must be the registered releaseOwner address");
+    vm.prank(randomAddress);
+    releaseGold.expire();
+  }
+
+  function test_ShouldSucceed_WhenCalledByReleaseOwner_WhenEXPIRATION_TIMEHasPassedAfterReleaseScheduleCompletion()
+    public
+  {
+    (uint256 releaseStartTime, , uint256 numReleasePeriods, uint256 releasePeriod, ) = releaseGold
+      .releaseSchedule();
+    vm.warp(
+      releaseStartTime + numReleasePeriods * releasePeriod + releaseGold.EXPIRATION_TIME() + 1
+    );
+    vm.prank(releaseOwner);
+    releaseGold.expire();
+
+    assertEq(releaseGold.isRevoked(), true);
+    (, , uint256 releasedBalanceAtRevoke, ) = releaseGold.revocationInfo();
+    assertEq(releasedBalanceAtRevoke, 0);
+  }
+
+  function test_ShouldAllowToRefundOfAllRemainingGold_WhenAnInstanceIsExpired_WhenBeneficiaryHasNotWithdrawnAnyBalanceYet_WhenEXPIRATION_TIMEHasPassedAfterReleaseScheduleCompletion()
+    public
+  {
+    (uint256 releaseStartTime, , uint256 numReleasePeriods, uint256 releasePeriod, ) = releaseGold
+      .releaseSchedule();
+    vm.warp(
+      releaseStartTime + numReleasePeriods * releasePeriod + releaseGold.EXPIRATION_TIME() + 1
+    );
+    vm.prank(releaseOwner);
+    releaseGold.expire();
+
+    uint256 balanceBefore = goldToken.balanceOf(initParams2._refundAddress);
+    vm.prank(releaseOwner);
+    releaseGold.refundAndFinalize();
+    uint256 balanceAfter = goldToken.balanceOf(initParams2._refundAddress);
+    assertEq(balanceAfter - balanceBefore, TOTAL_AMOUNT);
+  }
+
+  function test_ShouldSucceed_WhenAnInstanceIsExpired_WhenBeneficiaryHasNotWithdrawnAnyBalanceYet_WhenEXPIRATION_TIMEHasPassedAfterReleaseScheduleCompletionAndBeneficiaryHasWithdrawnSomeBalance()
```

Comment:
seems that two seperate tests were merged into 1.

It might be best to keep the granularity of the test or update the test title, as the test does not only test if it succeed, but that the contract marked as revoked, and that the release balance at revoke is set to 0.

Reply:
I split the test